src/v/iceberg/manifest_io.h

```diff
@@ -0,0 +1,111 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "base/outcome.h"
+#include "bytes/iobuf.h"
+#include "cloud_io/remote.h"
+#include "iceberg/manifest.h"
+#include "iceberg/manifest_list.h"
+#include "model/fundamental.h"
+#include "utils/named_type.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/util/noncopyable_function.hh>
+
+#include <system_error>
+
+namespace iceberg {
+
+using manifest_path
+  = named_type<std::filesystem::path, struct manifest_path_tag>;
+using manifest_list_path
+  = named_type<std::filesystem::path, struct manifest_path_tag>;
+
+class manifest_io {
+public:
+    enum class errc {
+        // The call has timed out. Callers should retry, if allowed.
+        timedout,
```

Comment:
Without specifying, this will get error code 0. I believe error code 0 is reserved for success, but I may be mistaken about that.

> But our enumeration must meet one condition: numeric value 0 must not represent an error situation. 0 represents a success in any error domain (category).

https://akrzemi1.wordpress.com/2017/07/12/your-own-error-code/

Reply:
Good catch. I don't want this to be used for the success case. Removing the std::error_code bits

Reply:
Looking around, it seems like another common pattern when you want to leave off the success case is to just specify the value for the first error case explicitly. I.e. `timedout = 1,`

Reply:
I'm not sure I follow -- is the concern what happens when you default initialize these? I thought the concern was more around std::error_code's interface, which is no longer included

Reply:
Done